aptos-move/framework/aptos-framework/sources/coin.move

```diff
@@ -528,9 +528,9 @@ module aptos_framework::coin {
     /// This function to resolve your problem, and for the gas optimize.
     /// Before:
     /// ```
-    /// let coin_mut = borrow_global_mut<StructName<CoinType>>(my_addr).coin;
-    /// let value_now = coin::value(&coin_mut);
-    /// let new_coin = coin::extract(&mut coin_mut, value_now - 100000000);
+    /// let coin_mut = borrow_global_mut<StructName<CoinType>>(my_addr);
```

Comment:
If there is a bug in the compiler can we please open an issue with a minimal example to reproduce the bug? In another context, I asked @e99243506bigplay already to do this, but did not got a reply.

The compiler has no issue in treating a `&mut T` as a `&T`. The following just works fine:

```
   struct C has key {
        v: u64
    }

    fun value(c: &C): u64 {
        c.v
    }

    fun foo():u64 acquires C {
        let mut_c = borrow_global_mut<C>(@42);
        let v = value(mut_c);
        v + 1
    }
```

I guess there was a different mistake in @e99243506bigplay code which let to they could not use the `&mut`? Borrow semantics is not easy. Perhaps asking with the according example which does not compile the community on Discord is a good idea?

As regards the gas optimization I do not think we want to make those APIs more heavy and confusing by adding new functions for marginal gas optimizations. Instead, we should (and will) invest soon in better code optimizations.

Reply:
which reply I didnt got received? let me test again.
Strange, I was thought it wont get compiled... How could be this result...

Reply:
> I guess there was a different mistake in @e99243506bigplay code which let to they could not use the &mut? Borrow semantics is not easy. Perhaps asking with the according example which does not compile the community on Discord is a good idea?

I will do in next time, thanks.